File: problem3.js
var checkPrime = function(prime) {
    for(var i = 2; i <= Math.sqrt(prime); i++) {
		if(prime % i === 0) return false;
	}
	return true;
};

var listPrimeFactors = function(num) {
	var factors = [];
	for(var i = 2; i <= Math.sqrt(num); i++) {
		if(num % i === 0 && checkPrime(i)) factors.push(i);
	}
	return factors;
};

var factors = listPrimeFactors(600851475143);

console.log(factors[factors.length - 1]);

File: problem3.rb
def check_prime prime
	i = 2
	while i <= Math.sqrt(prime)
		if prime % i == 0
			return false
		end
		i += 1
	end
	true
end

def list_prime_factors num
	factors = []
	i = 2
	while i <= num
		if num % i == 0 and check_prime i
			factors.push i
		end
		i += 1
	end
	factors
end

factors = list_prime_factors 69
puts factors
# puts factors.last